internal/search/searcher/client_grpc.go

```diff
@@ -80,7 +81,7 @@ func SearchGRPC(
 			return false, errors.Wrap(err, "failed to parse URL")
 		}
 
-		clientConn, err := grpc.Dial(parsed.Host, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		clientConn, err := grpc.DialContext(ctx, parsed.Host, append(grpcdefaults.DialOptions(), grpc.WithTransportCredentials(insecure.NewCredentials()))...)
```

Comment:
Incredibly minor nit, but it be nice to break this up into a few lines to keep the line length down:

```suggestion
		var opts []grpc.DialOption
		opts = append(opts, grpcdefaults.DialOptions()...)
		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
		
		clientConn, err := grpc.DialContext(ctx, parsed.Host, opts...)
```